packages/block-ingestor/src/index.ts

```diff
@@ -0,0 +1,76 @@
+import { PutBlocksReq } from "./api/put-blocks";
+import { BitcoinNetwork, type BlockQueueMessage } from "@gonative-cc/lib/bitcoin";
+
+export async function handleIngestBlocks(request: Request, env: Env): Promise<Response> {
+	const blocks = PutBlocksReq.decode(await request.arrayBuffer());
+	if (blocks.length === 0) {
+		return new Response("Empty block batch", { status: 400 });
+	}
+
+	const blockMetas = blocks.map((block) => {
+		const blockHash = block.block.getId();
+		const kvKey = `blocks:${block.network}:${blockHash}`;
+		return { block, blockHash, kvKey };
+	});
+
+	// Batch KV puts
+	await Promise.all(
+		blockMetas.map((meta) => env.BLOCK_STORE.put(meta.kvKey, meta.block.block.toBuffer())),
+	);
+
+	// Group messages by network
+	const messagesByNetwork: Partial<Record<BitcoinNetwork, BlockQueueMessage[]>> = {};
```

Comment:
Let's think: should we have a separate worker per network or not?

Reply:
I think the logic to distinguish the netowrk is very minimal, and its simpler to have just one worker